packages/amplify-category-function/src/provider-utils/awscloudformation/utils/convertLayersTypes.ts

```diff
@@ -0,0 +1,52 @@
+import { ExternalLayer, LambdaLayer, ProjectLayer } from 'amplify-function-plugin-interface';
+import { convertProjectLayer } from './layerArnConverter';
+
+const LAYER_ARN_KEY = 'Fn::Sub';
+
+export const convertProjectLayersToExternalLayers = (lambdaLayers: LambdaLayer[], envName: string): LambdaLayer[] => {
```

Comment:
based on the caller you can end up with `undefined` for the `lambdaLayers` parameter in case when a function does not have a layer, the caller has a `if (resourceName)` condition and if the code is not going in there then this parameter will be `undefined` that needs to be checked

Reply:
This is coming from 
```    if (Array.isArray(resourceParams.lambdaLayers) && resourceParams.lambdaLayers.length) {```

Added an Array check on lambdaLayers so that no undefined is passed